tensilelite/Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp

```diff
@@ -200,6 +201,59 @@ namespace Tensile
                 }
             };
 
+            struct SynchronizerSizeCheck
+                : public Predicate_CRTP<SynchronizerSizeCheck, ContractionProblemGemm>
+            {
+                enum
+                {
+                    HasIndex = true,
+                    HasValue = true
+                };
+                size_t             index;
+                std::array<int, 5> value;
+
+                SynchronizerSizeCheck() = default;
+                SynchronizerSizeCheck(size_t index, std::array<int, 5> value)
+                    : index(index)
+                    , value(value)
+                {
+                }
+
+                static std::string Type()
+                {
+                    return "SynchronizerSizeCheck";
+                }
+
+                virtual bool operator()(ContractionProblemGemm const& problem) const override
+                {
+                    // WorkGroup numbers x global write batch x Wave numbers
```

Comment:
global write batch => number of global write instruction

Reply:
modified